frontend/js/src/utils/coverArtCache.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable no-console */
+import localforage from "localforage";
+
+// Initialize IndexedDB
+const coverArtCache = localforage.createInstance({
+  name: "listenbrainz",
+  driver: [localforage.INDEXEDDB, localforage.LOCALSTORAGE],
+  storeName: "coverart",
+});
+
+const coverArtCacheExpiry = localforage.createInstance({
```

Comment:
I believe, one of the important factors of saving data in IndexedDB / LocalStorage is the cleanup process. What `removeAllExpiredCacheEntries` does is run over all the keys and check if the key has been expired or not.

Saving the expiration key as `${key}_expires_${hash}` makes this process slower as we'll have to index all keys, and filter down the keys of this regex and then process.

Reply:
```
interface CacheEntry {
  data: string;
  expiry: number
}
```

This approach is definitely better because a Single operation ensures data and expiry are always in sync, and makes the code simpler. BUT, when checking if data is expired, this forces you to deserialize the entire object which will increase memory usage and processing time.